packages/flutter/lib/src/material/focusable.dart

```diff
@@ -0,0 +1,297 @@
+  /// [FocusableThemeData] in the ambient [Theme].
+  ///
+  /// Defaults to true.
+  final bool showDecorations;
+
+  /// Defines the decoration to be drawn around the [child] when this widget
+  /// receives the focus.
+  ///
+  /// This defaults to the [FocusableThemeData.focusedDecoration] provided by
+  /// the ambient [Theme] in the context.
+  final Decoration focusedDecoration;
+
+  /// Defines the decoration to be drawn around the [child] when this widget
+  /// does not have the focus.
+  ///
+  /// This defaults to the [FocusableThemeData.unfocusedDecoration] provided by
+  /// the ambient [Theme] in the context.
+  final Decoration unfocusedDecoration;
+
+  /// Returns the [node] of the [Focusable] that most tightly encloses the given
+  /// [BuildContext].
+  ///
+  /// The [context] argument must not be null.
+  static FocusableNode of(BuildContext context) {
+    assert(context != null);
+    final _FocusableMarker marker = context.inheritFromWidgetOfExactType(_FocusableMarker);
+    return marker?.node ?? context.owner.focusManager.rootFocusable;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(StringProperty('debugName', debugLabel, defaultValue: null));
+    properties.add(DiagnosticsProperty<Decoration>('decoration', focusedDecoration));
+  }
+
+  @override
+  _FocusableState createState() => _FocusableState();
+}
+
+class _FocusableState extends State<Focusable> {
+  FocusableNode node;
+  bool _hasFocus;
+
+  @override
+  void initState() {
+    super.initState();
+    node = FocusableNode(
+      isScope: widget.isScope,
+      autofocus: widget.autofocus,
+      key: GlobalKey(debugLabel: widget.debugLabel),
+    );
+    _hasFocus = node.hasFocus;
+    node.addListener(_handleFocusChanged);
+  }
+
+  @override
+  void dispose() {
+    node.dispose();
+    super.dispose();
+  }
+
+  void _handleFocusChanged() {
+    if (_hasFocus != node.hasFocus) {
+      setState(() {
+        _hasFocus = node.hasFocus;
+      });
+      if (widget.onFocusChange != null) {
+        widget.onFocusChange(node.hasFocus);
+      }
+    }
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final FocusableNode newParent = Focusable.of(context);
+    newParent.reparent(node);
+  }
+
+  Decoration _getDecoration(BuildContext context) {
+    if (!widget.showDecorations) {
+      return null;
+    }
+    if (_hasFocus) {
+      return widget.focusedDecoration ?? Theme.of(context).focusableTheme.focusedDecoration;
+    } else {
+      return widget.unfocusedDecoration ?? Theme.of(context).focusableTheme.unfocusedDecoration;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return RawKeyboardListener(
+      focusNode: node,
+      onKey: (RawKeyEvent event) {
+        if (widget.onKey != null) {
+          widget.onKey(node, event);
+        }
+      },
+      child: Container(
```

Comment:
use an AnimatedContainer and expose the curve and duration

Reply:
maybe get the default curve and duration from the theme too

Reply:
Ooh, good idea.